lib/token.cpp

```diff
@@ -1695,119 +1695,171 @@ std::string Token::astStringZ3() const
 
 void Token::printValueFlow(bool xml, std::ostream &out) const
 {
+    std::string outs;
+
     int line = 0;
     if (xml)
-        out << "  <valueflow>" << std::endl;
+        outs += "  <valueflow>\n";
     else
-        out << "\n\n##Value flow" << std::endl;
+        outs += "\n\n##Value flow\n";
     for (const Token *tok = this; tok; tok = tok->next()) {
         const auto* const values = tok->mImpl->mValues;
         if (!values)
             continue;
         if (values->empty()) // Values might be removed by removeContradictions
             continue;
-        if (xml)
-            out << "    <values id=\"" << values << "\">" << std::endl;
-        else if (line != tok->linenr())
-            out << "Line " << tok->linenr() << std::endl;
+        if (xml) {
+            outs += "    <values id=\"";
+            outs += ptr_to_string(values);
+            outs +=  "\">";
+            outs += '\n';
+        }
+        else if (line != tok->linenr()) {
+            outs += "Line ";
+            outs += std::to_string(tok->linenr());
+            outs += '\n';
+        }
         line = tok->linenr();
         if (!xml) {
             ValueFlow::Value::ValueKind valueKind = values->front().valueKind;
             const bool same = std::all_of(values->begin(), values->end(), [&](const ValueFlow::Value& value) {
                 return value.valueKind == valueKind;
             });
-            out << "  " << tok->str() << " ";
+            outs += "  ";
```

Comment:
nit: we can use `' '` here

Reply:
Probably - if it were a single character... but I guess that is just a typo.

Reply:
It seems the code using a `char` is much slower similar to the stream insertion case I mentioned in another comment. Will investigate but that is out-of-scope of this PR.

Reply:
yes it should have been a single character that was a typo :+1: